### ISSUE TITLE
Solving LQR problems with cross term matrix

### ISSUE THREAD

**Issue by 48cfu** [ApolloAuto/apollo#9894]
Name: [SolveLQRProblem](https://github.com/ApolloAuto/apollo/blob/master/modules/common/math/linear_quadratic_regulator.h) **
about: Overriding the function SolveLQRProblem with an additional parameter M, which indicates the cross term matrix when the loss function to minimize is of the form x'Qx + u'Ru + 2x'Mu. (x' and u' indicates x and u transposed)

**Is your feature request related to a problem? Please describe.**
I was trying to solve an LQR problem with cross term and I noticed that the current [SolveLQRProblem](https://raw.githubusercontent.com/ApolloAuto/apollo/master/modules/common/math/linear_quadratic_regulator.cc) only handles the case where the loss function is of the form x'Qx + u'Ru

**Describe the solution you'd like**
I would like to be assigned to this issue and contribute to this project by implementing the overloaded function SolveLQRProblem which computes the optimal gain also in presence of the cross term matrix M.

**Additional context**
Furthermore, I was wondering why in the current implementation of [SolveLQRProblem](https://raw.githubusercontent.com/ApolloAuto/apollo/master/modules/common/math/linear_quadratic_regulator.cc) the feasibility of the solution is not check. That is,  (A, B) stabilizable and (Q, A) detectable. For the _proposed_ function_ this conditions would then be (A, B) stabilizable and (Q - N R^-1 N', A - B R^-1 N') detectable

**Comment by Capri2014** [comment]
@48cfu Yes, please do so! This would help in the Lon-lat coupled modeling. Thanks. -- Also for the feasibility check, I think it makes sense too, only thing is I would recommend to show AERROR instead of AFATAL during runtime.

**Comment by changshiwang** [comment]
Hellor
I am a china automation engineer.I wish to  know the property of cross term matrix M. thank you
